takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/AbstractCompiler.java

```diff
@@ -55,6 +55,8 @@
 
   private boolean showWarnings;
 
+  private boolean failOnError = true;
```

Comment:
individual compiler backends do not care about `failOnError` behaviour. please remove this field and corresponding getter/setter methods.